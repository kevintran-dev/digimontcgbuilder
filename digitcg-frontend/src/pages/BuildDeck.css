/* .create-new-button {
    display: inline-block;
}

.delete-btn {
    display: inline-block;

} */
body {
    background-color: rgb(213, 208, 208);
}

.deck-btns {
    display: inline-block;
}

.loadBtn {
    display: inline-block;
    margin: 0.5em
}

.searchForm {
    min-width: 100%;
}

.select-options {
    display: inline-block;
    padding-left: 2%;
    padding-right: 2%;
    padding-top: 2%;
    padding-bottom: 8%;
}