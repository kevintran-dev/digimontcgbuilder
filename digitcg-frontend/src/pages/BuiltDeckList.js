import React from 'react'
import BuiltCard from './BuiltCard'

const BuiltDeckList = ({ savedDeck, removeFromDeck }) => {

    return (
        <div className='BuiltDeck-list'>
            {Object.keys(savedDeck).map(data => (
                <BuiltCard
                    key={savedDeck[data].id}
                    id={savedDeck[data].id}
                    image_url={savedDeck[data].src}
                    level={savedDeck[data].level}
                    type={savedDeck[data].type}
                    cardnumber={savedDeck[data].cardnumber}
                    removeFromDeck={removeFromDeck}
                />
            ))}
        </div>
    )
}

export default BuiltDeckList