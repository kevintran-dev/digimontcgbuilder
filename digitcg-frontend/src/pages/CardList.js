import React from 'react'
import DigiCard from './DigiCard'

const CardList = ({ removeFromDeck, deckDiv, setDeckDiv, cardData, deckInfo, setdeckInfo }) => {
    console.log(cardData)
    return (
        <div className='DigiCard-list'>
            {Object.keys(cardData).map(data => (
                <DigiCard
                    key={cardData[data].cardnumber}
                    cardnumber={cardData[data].cardnumber}
                    artist={cardData[data].artist}
                    attribute={cardData[data].attribute}
                    card_sets={cardData[data].card_sets}
                    cardrarity={cardData[data].cardrarity}
                    color={cardData[data].color}
                    digi_type={cardData[data].digi_type}
                    dp={cardData[data].dp}
                    evolution_cost={cardData[data].evolution_cost}
                    image_url={cardData[data].image_url}
                    level={cardData[data].level}
                    maineffect={cardData[data].maineffect}
                    name={cardData[data].name}
                    play_cost={cardData[data].play_cost}
                    set_name={cardData[data].set_name}
                    sourceeffect={cardData[data].sourceeffect}
                    stage={cardData[data].stage}
                    type={cardData[data].type}
                    deckInfo={deckInfo}
                    setdeckInfo={setdeckInfo}
                    deckDiv={deckDiv}
                    setDeckDiv={setDeckDiv}
                    removeFromDeck={removeFromDeck}
                />
            ))}
        </div>
    )
}

export default CardList