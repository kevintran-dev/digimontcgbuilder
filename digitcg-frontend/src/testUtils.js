import React from "react";
import UserContext from "./auth/UserContext";

const UserProvider =
    ({ children, currentUsername = "testuser", currentUser = [], update = false }) => (
        <UserContext.Provider value={{ currentUser, currentUsername, update }}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };

