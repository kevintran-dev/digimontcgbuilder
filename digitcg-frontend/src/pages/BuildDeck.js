import React, { useEffect, useState, useContext } from 'react'
import NameSearchForm from '../forms/NameSearchForm';
import axios from "axios";
import './DigiCard.css'
import CardList from './CardList';
import DigiBackendApi from '../api/DigiBackendApi';
import UserContext from "../auth/UserContext";
import BuiltDeckList from './BuiltDeckList';
import LoadDecksForm from '../forms/LoadDecksForm';
import DeleteDecksForm from '../forms/DeleteDecksForm';
import CreateDecksForm from '../forms/CreateDecksForm';
import UpdateDecksForm from '../forms/UpdateDecksForm';
import './BuildDeck.css'


const BuildDeck = () => {
    const [deckInfo, setdeckInfo] = useState([]);
    const [targetId, setTargetId] = useState("");
    const CARD_URL = "https://digimoncard.io/api-public/search.php?";
    const [cardRes, setCardRes] = useState("")
    const [deckDiv, setDeckDiv] = useState()
    const [levelCountDiv, setLevelCountDiv] = useState()
    const { currentUser, setCurrentUser } = useContext(UserContext)
    const [deckSaved, setDeckSaved] = useState(false)
    const [currentDeckId, setCurrentDeckId] = useState()
    const [currentDeckName, setCurrentDeckName] = useState()

    const [notFound, setNotFound] = useState(false);
    const [builtDeckNotFound, setBuiltDeckNotFound] = useState(false);
    const [builtDeckFound, setBuiltDeckFound] = useState(false);

    const [updateState, setUpdateState] = useState(false);

    useEffect(function () {
        setDeckSaved(false);

        if (currentUser.length > 0) {
            setBuiltDeckFound(true)
        }
        if (currentUser.length === 0) {
            setBuiltDeckFound(false)
        }

        if (!builtDeckNotFound) {
            setBuiltDeckNotFound(true)
        }
        if (updateState) {
            removeFromState()
            setUpdateState(false)
        }
    }, [deckInfo, updateState, deckSaved, currentDeckId])

    async function search(params) {
        try {
            let res = await axios.get(CARD_URL, { params });
            setCardRes(res.data)
            setNotFound(false);
            return res.data
        }
        catch {
            console.log("Not found");
            setNotFound(true);
        }
    }



    // async function searchForAllDecks() {
    //     try {
    //         let decks = await DigiBackendApi.getAllDecks();
    //         // setSavedDecks(decks)
    //     }
    //     catch {
    //         console.log("Not found");
    //     }
    // }

    // feature not added
    // const addToDeckInfoDiv = () => {
    //     const deckInfoP = document.querySelector(".built-deck-info");

    //     let levelCount = deckInfo.reduce((acc, cur) => {
    //         const level = cur.level;
    //         if (acc[level]) {
    //             acc[level]++;
    //         }
    //         else {
    //             acc[level] = 1;
    //         } return acc;
    //     }, {});

    //     let str = Object.keys(levelCount).map(key => `Level ${key}: ${levelCount[key]}`);
    //     deckInfoP.innerText = str.join(", ");
    //     console.log("new levels added to p div", deckInfoP)
    //     setLevelCountDiv(deckInfoP)
    // }


    const removeFromDeck = (e) => {
        let cardId = e.target.id;
        setTargetId(cardId);
        setUpdateState(true)

    }
    function remove(array, value) {
        return array.filter((element) => element.id !== value)
    }
    const removeFromState = () => {
        let newCopy = [...deckInfo]
        let newDeckInfo = remove(newCopy, targetId)
        setdeckInfo(newDeckInfo)
    }




    return (
        <>
            <h2>Deck Editor <img src="https://i.pinimg.com/originals/f7/a5/fd/f7a5fdaf66c24f579b8e68a6798f593c.gif" alt="calumon" width="60px" /></h2>

            <div className='searchForm'>
                <NameSearchForm searchFor={search} />
            </div>

            {builtDeckFound
                ?
                currentUser.map((deck, index) => (<LoadDecksForm key={index} deck={deck} deckInfo={deckInfo} setdeckInfo={setdeckInfo} setCurrentDeckId={setCurrentDeckId} setCurrentDeckName={setCurrentDeckName} />))
                :
                <div className='LoadDecksForm'>
                </div>
            }


            {!notFound
                ?

                <CardList removeFromDeck={removeFromDeck} deckDiv={deckDiv} setDeckDiv={setDeckDiv} cardData={cardRes} deckInfo={deckInfo} setdeckInfo={setdeckInfo} />

                : <p className="lead">Sorry, no results were found!</p>
            }

            <div className="built-deck">
                <p className="built-deck-info"></p>
                <div className="deck-btns">
                    <CreateDecksForm deckInfo={deckInfo} setDeckSaved={setDeckSaved} currentDeckId={currentDeckId} currentDeckName={currentDeckName} />
                    <DeleteDecksForm deckInfo={deckInfo} setdeckInfo={setdeckInfo} setCurrentDeckId={setCurrentDeckId} currentDeckId={currentDeckId} />
                </div>
                <BuiltDeckList savedDeck={deckInfo} removeFromDeck={removeFromDeck} />

            </div>
        </>

    )
}

export default BuildDeck