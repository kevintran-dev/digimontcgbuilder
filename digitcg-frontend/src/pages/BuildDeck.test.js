import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router"
import { UserProvider } from "../testUtils";
import BuildDeck from "./BuildDeck";
import CardList from "./CardList";
import BuiltDeckList from "./BuiltDeckList";

it("renders without crashing", function () {
    let card = [{ artist: "ryuda_1", attribute: "Vaccine", card_sets: ["ST-7: Starter Deck Gallantmon", "BT01-03: Release Special Booster Ver.1.0", "BTC-01: Booster Ultimate Evolution", "ST-7: Starter Deck Gallantmon", "BT-01: Booster New Evolution"], cardnumber: "BT1-009", cardrarity: "Common", color: "Red", digi_type: "Mini Dragon", dp: 3000, evolution_cost: 0, image_url: "https://images.digimoncard.io/images/cards/BT1-009.jpg", level: 3, maineffect: null, name: "Monodramon", play_cost: 2, set_name: "ST-7: Starter Deck Gallantmon", soureeffect: null, stage: "Rookie", type: "Digimon" }]
    let savedDeck = [{ "cardnumber": "BT11-001", "level": 2, "type": "Digi-Egg", "src": "https://images.digimoncard.io/images/cards/BT11-001.jpg", "id": "dca81367-e025-4214-b04e-e86a0c0b6f9f" }, { "cardnumber": "BT11-002", "level": 2, "type": "Digi-Egg", "src": "https://images.digimoncard.io/images/cards/BT11-002.jpg", "id": "009d02a5-acee-4379-aa94-f3a70f06fcf8" }, { "cardnumber": "BT11-003", "level": 2, "type": "Digi-Egg", "src": "https://images.digimoncard.io/images/cards/BT11-003.jpg", "id": "cfeafe7f-2a3f-4036-9e18-620b4605b59f" }, { "cardnumber": "BT11-004", "level": 2, "type": "Digi-Egg", "src": "https://images.digimoncard.io/images/cards/BT11-004.jpg", "id": "0329a814-bf56-4088-bc58-a56b65d5534c" }, { "cardnumber": "BT11-005", "level": 2, "type": "Digi-Egg", "src": "https://images.digimoncard.io/images/cards/BT11-005.jpg", "id": "10defd34-d4a5-4250-8a78-0dfa2eca1a4a" }]
    render(<CardList cardData={card} />);
    render(<BuiltDeckList savedDeck={savedDeck} />)
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <BuildDeck />
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})

