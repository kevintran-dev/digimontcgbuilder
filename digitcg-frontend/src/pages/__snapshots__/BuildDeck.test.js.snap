// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Deck Editor
  </h2>
  <div
    class="searchForm"
  >
    <form>
      <div>
        <label
          for="n"
        />
        <input
          id="n"
          name="n"
          placeholder="Enter Digimon name here.."
          type="text"
          value=""
        />
      </div>
      <div
        class="select-options"
      >
        <label>
           Color
          <select
            name="color"
          >
            <option
              disabled=""
              value="true"
            >
              -- select an option --
            </option>
            <option
              value="black"
            >
              Black
            </option>
            <option
              value="blue"
            >
              Blue
            </option>
            <option
              value="colorless"
            >
              Colorless
            </option>
            <option
              value="green"
            >
              Green
            </option>
            <option
              value="red"
            >
              Red
            </option>
            <option
              value="white"
            >
              White
            </option>
            <option
              value="yellow"
            >
              Yellow
            </option>
          </select>
        </label>
      </div>
      <div
        class="select-options"
      >
        <label>
           Type
          <select
            name="card-type"
          >
            <option
              disabled=""
              value="true"
            >
              -- select an option --
            </option>
            <option
              value="digimon"
            >
              Digimon
            </option>
            <option
              value="option"
            >
              Option
            </option>
            <option
              value="tamer"
            >
              Tamer
            </option>
            <option
              value="digi-egg"
            >
              Digi-Egg
            </option>
          </select>
        </label>
      </div>
      <div
        class="select-options"
      >
        <label>
           Atrribute
          <select
            name="attribute"
          >
            <option
              disabled=""
              value="true"
            >
              -- select an option --
            </option>
            <option
              value="data"
            >
              Data
            </option>
            <option
              value="free"
            >
              Free
            </option>
            <option
              value="vaccine"
            >
              Vaccine
            </option>
            <option
              value="variable"
            >
              Variable
            </option>
            <option
              value="virus"
            >
              Virus
            </option>
          </select>
        </label>
      </div>
      <div
        class="select-options"
      >
        <label>
           Stage
          <select
            name="stage"
          >
            <option
              disabled=""
              value="true"
            >
              -- select an option --
            </option>
            <option
              value="baby"
            >
              Baby
            </option>
            <option
              value="In-Training"
            >
              In-Training
            </option>
            <option
              value="rookie"
            >
              Rookie
            </option>
            <option
              value="champion"
            >
              Champion
            </option>
            <option
              value="ultimate"
            >
              Ultimate
            </option>
            <option
              value="mega"
            >
              Mega
            </option>
          </select>
        </label>
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
      <button>
        Clear Selections
      </button>
    </form>
  </div>
  <div
    class="LoadDecksForm"
  />
  <div
    class="DigiCard-list"
  />
  <div
    class="built-deck"
  >
    <p
      class="built-deck-info"
    />
    <div
      class="deck-btns"
    >
      <div
        class="create-new-button"
      >
        <form>
          <div>
            <label
              for="DeckName"
            />
            <input
              id="submit"
              name="DeckName"
              placeholder="Name your deck!"
              type="text"
              value=""
            />
          </div>
          <button
            type="submit"
          >
            Create New Deck
          </button>
        </form>
        <div>
          <form>
            <button
              type="submit"
            >
              Update Deck
            </button>
          </form>
        </div>
      </div>
      <div
        class="delete-btn"
      >
        <form>
          <button
            type="submit"
          >
            Delete Deck
          </button>
        </form>
      </div>
    </div>
    <div
      class="BuiltDeck-list"
    />
  </div>
</DocumentFragment>
`;
